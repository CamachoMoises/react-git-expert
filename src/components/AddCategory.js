import React, { useState } from 'react';

export const AddCategory = () => {
    const [inputValue, setInputValue] = useState('New category')

    const handleInputChange=(e)=>{
        setInputValue(e.target.value)
    }
    const handleSubmit= (e)=>{
        e.preventDefault()
        console.log('submitted');
    }   
	return (
		<form onSubmit={handleSubmit}>
			<h2> Add Category {inputValue} </h2> 
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />

		</form>
	);
};
