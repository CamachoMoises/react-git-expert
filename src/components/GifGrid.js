import React, { useEffect, useState } from 'react';

export const GifGrid = ({ category }) => {
	const [images, setImages] = useState([]);
	useEffect(() => {
		getGif();
	}, []);

	const getGif = async () => {
		const url = 'https://api.giphy.com/v1/gifs/search?q=Speed+Racer&limit=10&api_key=wnzsNoA0er1NfL4znwh2pMAqRWa5BTi2';
		const resp = await fetch(url);
		const { data } = await resp.json();
		const gifs = data.map((gif) => {
			return {
				id: gif.id,
				title: gif.title,
				url: gif.images?.downsized_small.mp4,
			};
		});
		console.log(gifs);
		setImages(gifs);
	};

	return (
		<div>
			<h3> {category} </h3>
			<ol>
				{images.map(({id, title}) => (
					<li key={id}> {title}</li>
				))}
			</ol>
		</div>
	);
};
