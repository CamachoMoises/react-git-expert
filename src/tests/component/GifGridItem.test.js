import React from 'react';
import { GifGridItem } from '../../components/GifGridItem';
import { shallow } from 'enzyme';

describe('test in GifGridItem.js file', () => {
	const title = 'test title';
	const url = 'http://localhost.something.jpg';
	const wrapper = shallow(<GifGridItem title={title} url={url} />);

	test('should show the component GridItem', () => {
		expect(wrapper).toMatchSnapshot();
	});
	test('it should have a paragraph with the title', () => {
		const p = wrapper.find('p');
		expect(p.text().trim()).toBe(title);
	});
	test('it should have a image from url and alt of props', () => {
		const img = wrapper.find('img');
		console.log('Props', img.prop('src'));
		expect(img.props().src).toBe(url);
        expect(img.props().alt).toBe(title);
	});
    test('main tag should have in class animate__fadeIn', () => {
        const div = wrapper.find('div');
		// console.log('Props', div.prop('className'));
        const className= div.prop('className')
        // console.log('Watch out!', className.includes('animate__fadeIn') );
        expect(className.includes('animate__fadeIn')).toBe(true);

    })
    
});
