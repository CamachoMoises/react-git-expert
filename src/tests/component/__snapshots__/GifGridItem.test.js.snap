// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in GifGridItem.js file should show the component GridItem 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="test title"
    src="http://localhost.something.jpg"
  />
  <p>
    test title
  </p>
</div>
`;
